src/ts/components/subtitlesettings/backgroundcolorselectbox.ts

```diff
@@ -0,0 +1,48 @@
+import {SelectBox} from '../selectbox';
+import {ListSelectorConfig} from '../listselector';
+import {UIInstanceManager} from '../../uimanager';
+import {SubtitleOverlay} from '../subtitleoverlay'
+import {ColorUtils, StorageUtils} from '../../utils';
+
+/**
+ * A select box providing a selection of different background colors.
+ */
+export class BackgroundColorSelectBox extends SelectBox {
+
+  private overlay: SubtitleOverlay;
+
+  constructor(config: ListSelectorConfig = {}, overlay: SubtitleOverlay) {
```

Comment:
Components should receive all configuration properties through the `Config` object, which means you should also pass the `SubtitleOverlay` as config property. You can make an abstract `SubtitlePropertySelectBox` with a `SubtitlePropertySelectBoxConfig` and use that as a base class for all select boxes.

```ts
export interface SubtitlePropertySelectBoxConfig extends ListSelectorConfig { 
  subtitleOverlay: SubtitleOverlay; 
}

export abstract class  SubtitlePropertySelectBox extends SelectBox {

  protected subtitleOverlay: SubtitleOverlay;

  constructor(config: SubtitlePropertySelectBoxConfig) {
    super(config);
    this.subtitleOverlay = config.subtitleOverlay;
    ...
```